package.json

```diff
@@ -18,19 +18,20 @@
   "types": "lib/peg.d.ts",
   "bin": "bin/peggy",
   "scripts": {
-    "pretest": "npm run parser",
-    "test": "npm run test:bare",
-    "test:bare": "mocha --recursive test --exclude test/browser.stub.js --reporter min",
+    "clean": "rimraf -f build && rimraf -f browser && mkdir build && cd build && mkdir ts && mkdir rollup && mkdir terser && cd .. && mkdir browser",
```

Comment:
`mkdir -p` ?

Reply:
Also, can we just put all of the ephemeral build products in one directory?

Reply:
> Also, can we just put all of the ephemeral build products in one directory?

they are all in `build/`

the reason they're not in the same subdirectory is that this is easier for debugging, this allows you to dovetail multiple build paths, this is easier for static analysis, and this makes it easier to add later stages

if you want everything in just-build instead of `build/stepname` i can switch it though

Reply:
> mkdir -p

not portable to windows, behavior is buggy on old macs

Reply:
> if you want everything in just-build instead of `build/stepname` i can switch it though

Yes, that would work for me.  Thank you.

Reply:
for the record, you also seemed to support 

> Ideally, each step should be in buildable state, i.e. lint + build without errors

which is exactly what I will now remove, because those steps are in subdirectories, where they can be compared, rather than just leaving you with the most recent one

Reply:
This is also going to cause a lot of hay

Most of these tools batch work, and produce files with the same name as their inputs

I still need the subdirectories; I just collapse them at the end (it's not entirely clear why)

Reply:
Ancillary build step `tidy` now cleans intermediate build products